source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -518,6 +518,14 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
 
   server_transformation_ = config.server_header_transformation();
 
+  if (!config.scheme_header_transformation().scheme_to_overwrite().empty()) {
+    scheme_ = config.scheme_header_transformation().scheme_to_overwrite();
+    if (scheme_.value() != Http::Headers::get().SchemeValues.Https &&
+        scheme_.value() != Http::Headers::get().SchemeValues.Http) {
+      throw EnvoyException(fmt::format("Invalid scheme: {}", scheme_.value()));
```

Comment:
nit: can you make this error message a bit more obvious of what field it refers to (though per above I think you can have PGV check this)